packages/block-library/src/navigation/edit/index.js

```diff
@@ -152,6 +153,37 @@ function Navigation( {
 			name: 'block-library/core/navigation/create',
 		} );
 
+	const generatedTemplatePartHierarchySlug = useSelect(
+		( select ) => {
+			// Use the lack of a clientId as an opportunity to bypass the rest
+			// of this hook.
+			if ( ! clientId ) {
+				return;
+			}
+
+			const { getBlock, getBlockParentsByBlockName } =
+				select( blockEditorStore );
+
+			const withAscendingResults = true;
+			const parentTemplatePartClientIds = getBlockParentsByBlockName(
+				clientId,
+				'core/template-part',
+				withAscendingResults
+			);
+
+			let result = parentTemplatePartClientIds.map(
+				( templatePartClientId ) => {
+					return getBlock( templatePartClientId )?.attributes?.slug;
+				}
+			);
+
+			result = result.join( '-' ).trim();
```

Comment:
```suggestion
			result = result.join( '--' ).trim();
```
Might be best to use `--` to make it easier to know where the individual parts have been stitched together.

Reply:
Looks like WP collapses the double dash. I've used a hardcoded string for now but that might need to change.